### ISSUE TITLE
Screenshot is not captured on hook failure

### ISSUE THREAD

**Issue by Apoorva-GA** [getgauge/gauge-js#149]
_From @sguptatw on March 29, 2018 3:6_

**Expected behavior**
The screenshot should be displayed on failure

**Actual behavior**
Screenshot is not displayed on failure

**Steps to replicate**
* Create a `gauge-js` project
* Add a new step
* Give the default implementation
* Run the specs `gauge run specs`
* Open the html-report
> No screenshot is displayed

**Version**
```
Gauge version: 0.9.8.nightly-2018-03-28
Commit Hash: 66fc822

Plugins
-------
html-report (4.0.4.nightly-2018-03-28)
js (2.2.0.nightly-2018-03-28)
```

_Copied from original issue: getgauge/html-report#172_

**Comment by Apoorva-GA** [comment]
_From @sguptatw on March 29, 2018 3:12_

The screenshot version is `screenshot (0.0.1)`

**Comment by Apoorva-GA** [comment]
_From @sguptatw on March 29, 2018 4:58_

There are 2 issues in this
- In a step, in JS project - when `throw new Error('something')` is used, screenshot is captured. `throw 'something'` Screenshot is not captured - https://github.com/getgauge/gauge-js/issues/134
- The current issue is to track `When there is an error in hooks, screenshot is not displayed.`

**Comment by Apoorva-GA** [comment]
I think the given `steps to replicate` and the next comment which says `When there is an error in hooks, screenshot is not displayed` are two different things.

Whats the actual issue here?

**Comment by Apoorva-GA** [comment]
_From @sguptatw on May 21, 2018 8:14_

The actual issue is `When there is an error in the hooks, screenshot is not displayed`

**Steps to replicate**

* Create a gauge-js project
* Add a new hook implementation
* Give the implementation with error
* Run the specs gauge run specs
* Open the html-report

**Comment by Apoorva-GA** [comment]
Fix should be available in gauge-js Nightly >= 21/05/18

**Comment by sguptatw** [comment]
**Expected behavior**
Screenshot should be available on failure in hook

**Actual behavior**
Screenshot is not available in hook

**Steps to replicate**
* Project [js_basic.zip](https://github.com/getgauge/gauge-js/files/2593804/js_basic.zip)
> screenshot_on_failure = true
* run the specs `gauge run`
* Open the HTML-report
> The screenshot is not available for hook failure

**Version**
```
Gauge version: 1.0.4.nightly-2018-11-12
Commit Hash: a60ffbf

Plugins
-------
js (2.3.4.nightly-2018-11-13)
```

**Comment by sguptatw** [comment]
Blocker - The latest available gauge nightly is `Gauge/Nightly/1.0.4.nightly-2018-11-26` where as required nightly is >= 29-11-2018
